buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -389,16 +389,132 @@ private static long compareUintBigEndianB(
         return 0;
     }
 
+    private static final class SWARByteSearch {
```

Comment:
I wonder if this class would make sense as a top-level ByteSearch class, and adopt more methods.

Reply:
Yes yes and yes :)